### PR TITLE
Fix:(issue_1206) Default value shouldnt depend on env variable or val…

### LINKED ISSUE #1206: Flag default value is overwritten by environment in help output.
## my urfave/cli version is

v2.3.0

## Describe the bug

Flag default value should never change in help output.

## To reproduce

Build the following program against v2.3.0.

```
package main

import (
	"github.com/urfave/cli/v2"
	"os"
)

func main() {
	app := cli.NewApp()
	app.Name = "myapp"
	app.Flags = []cli.Flag{
		&cli.StringFlag{
			Name:        "myflag",
			Usage:       "my flag is my flag",
			EnvVars:     []string{"MYFLAG"},
			Value:       "defval",
		},
  }
  app.Run(os.Args)
}
```

## Observed behavior

What did you do?

```
MYFLAG=foo ./app --help
```

What did you see happen immediately after the reproduction steps above?

```
NAME:
   myapp - A new cli application

USAGE:
   app [global options] command [command options] [arguments...]

COMMANDS:
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --myflag value  my flag is my flag (default: "foo") [$MYFLAG]
   --help, -h      show help (default: false)
```

What did you expect to see?

```
[...]
GLOBAL OPTIONS:
   --myflag value  my flag is my flag (default: "defval") [$MYFLAG]
   --help, -h      show help (default: false)
```

## Expected behavior

Default value should never change (in help output).

### PR DESCRIPTION
**Pull request by dearchap**
…ue set from cmdline

## What type of PR is this?

_(REQUIRED)_

- bug

## What this PR does / why we need it:

_(REQUIRED)_

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes #1206 
## Special notes for your reviewer:

_(fill-in or delete this section)_

## Testing

make test

_(fill-in or delete this section)_

## Release Notes

_(REQUIRED)_

```release-note

```

### CODE REVIEW

File: flag_test.go [hunk 1]
```diff
@@ -859,16 +847,21 @@ func TestInt64FlagWithEnvVarHelpOutput(t *testing.T) {
...
+		expectedSuffix := suffixForEnv("APP_BAR")
 		if !strings.HasSuffix(output, expectedSuffix) {
 			t.Errorf("%s does not end with"+expectedSuffix, output)
 		}
 	}
 }
 
+func suffixForEnv(s string) string {
+	expectedSuffix := fmt.Sprintf(" [$%s]", s)
+	if runtime.GOOS == "windows" {
```
Comment (julian7) [inline flag_test.go:859]: can we do this with [constraints](https://pkg.go.dev/cmd/go#hdr-Build_constraints)? Like, having a file like `suffix_windows.go`:

```go
//go:build windows
package cli

import "fmt"

func suffixForEnv(s string) string {
        return fmt.Sprintf(" [%%%s%%]", s)
}
```

and another one called like `suffix_nonwindows.go`:

```go
//go:build !windows
package cli

import "fmt"

func suffixForEnv(s string) string {
        return fmt.Sprintf(" [$%s]", s)
}
```

**Review by julian7 — CHANGES_REQUESTED** [review]

File: flag_test.go (hunk 1 above)
Comment (dearchap) [reply flag_test.go]: Interesting. Yes we could do that. However I found a function in the code which does the suffix stuff already

**Comment by dearchap** [comment]
@julian7 What do you think of current implementation ?

**Comment by dearchap** [comment]
Merging as all requirements are satisfied

File: flag_test.go (hunk 1 above)
Comment (meatballhat) [reply flag_test.go]: Worth considering how this relates to https://github.com/urfave/cli/issues/1288 ?

**Pull request merged**